submissions/KostiantynOkhorzin/frogger-game/app.js

```diff
@@ -0,0 +1,128 @@
+const config = {
+    canvas: {
+        width: 505,
+        blockWidth: 101,
+        blockHeight: 80,
+    },
+    enemy: {
+        minSpeed: 100,
+        maxSpeed: 300,
+        initialPositionOnX: -101,
+        initialPositionOnY: {
+            firstLine: 80,
+            secondLine: 160,
+            thirdLine: 240
+        },
+        numberOfEnemy: {
+            firstLine: 1,
+            secondLine: 2,
+            thirdLine: 1
+        }
+    },
+    player: {
+        initialPositionOnX: 202,
+        initialPositionOnY: 400
+    }
+};
+
+class Enemy {
+    constructor(y) {
+        this.x = config.enemy.initialPositionOnX;
+        this.y = y;
+        this.speed = this.randomSpeed();
+        this.sprite = 'images/enemy-bug.png';
+    }
+
+    randomSpeed() {
+        return Math.round(Math.random() * (config.enemy.maxSpeed - config.enemy.minSpeed) + config.enemy.minSpeed);
+    }
+
+    checkCollision() {
+        if(player.x < this.x + config.canvas.blockWidth &&
+            player.x + config.canvas.blockWidth > this.x &&
+            player.y < this.y + config.canvas.blockHeight && 
+            config.canvas.blockHeight + player.y > this.y) {
+            player.x = config.player.initialPositionOnX;
+            player.y = config.player.initialPositionOnY;
```

Comment:
Classes should not refer to any global variables, like global variable `player`, which is an instance of `Player` class (referring to global constants and globals provided by the gaming platform like Resources is OK); Hint: pass `Player` instance as an argument to every enemy